toolchain/check/deduce.cpp

```diff
@@ -84,121 +138,111 @@ static auto NoteGenericHere(Context& context, SemIR::GenericId generic_id,
   diag.Note(context.generics().Get(generic_id).decl_id, DeductionGenericHere);
 }
 
-auto DeduceGenericCallArguments(
-    Context& context, SemIR::LocId loc_id, SemIR::GenericId generic_id,
-    SemIR::SpecificId enclosing_specific_id,
-    [[maybe_unused]] SemIR::InstBlockId implicit_params_id,
-    SemIR::InstBlockId params_id, [[maybe_unused]] SemIR::InstId self_id,
-    llvm::ArrayRef<SemIR::InstId> arg_ids) -> SemIR::SpecificId {
-  DeductionWorklist worklist(context);
-
-  llvm::SmallVector<SemIR::InstId> result_arg_ids;
-  llvm::SmallVector<Substitution> substitutions;
-
+auto DeductionContext::AddEnclosingArgs(SemIR::SpecificId enclosing_specific_id)
+    -> void {
+  if (!enclosing_specific_id.is_valid()) {
+    return;
```

Comment:
It might make the logical structure a little clearer if we set `first_deduced_index_` to 0 here, rather than in the default member initializer; that way this method is unconditionally responsible for initializing `first_deduced_index_`.

Reply:
The type doesn't have a default constructor. But after inlining this into the constructor, it's at least all done in the same function now.